pyccel/ast/core.py

```diff
@@ -237,9 +237,13 @@ def __init__(self, *args):
                 self._precision = max(a.precision for a in integers)
             else:
                 raise TypeError('cannot determine the type of {}'.format(self))
-            
+
+            ranks  = [a.rank for a in args]
             shapes = [a.shape for a in args]
 
+            if None in ranks:
+                self._rank  = None
+                self._shape = None
             if all(not (sh is None or isinstance(sh, PyccelArraySize)) for tup in shapes for sh in tup):
```

Comment:
It appears that `tup` could be `None` here, in which case we get the error
```
TypeError: 'NoneType' object is not iterable
```

Reply:
Yes. When `tup` is `None` then so is `ranks`. I missed the `if`->`elif` on the line below when rebasing